stock_release_channel_shipment_advice/models/stock_release_channel.py

```diff
@@ -0,0 +1,152 @@
+            "context": self.env.context,
+        }
+
+    @api.depends("shipment_planning_method", "picking_to_plan_ids")
+    def _compute_can_plan_shipment(self):
+        for rec in self:
+            rec.can_plan_shipment = (
+                rec.shipment_planning_method != "none" and rec.picking_to_plan_ids
+            )
+
+    @api.depends("picking_ids", "picking_ids.planned_shipment_advice_id")
+    def _compute_picking_to_plan_ids(self):
+        for rec in self:
+            rec.picking_to_plan_ids = rec.picking_ids.filtered(
+                lambda p: not p.planned_shipment_advice_id and p.state == "assigned"
+            )
+
+    def button_plan_shipment(self):
+        self.ensure_one()
+        if not self.can_plan_shipment:
+            raise UserError(_("Shipment planning not allowed"))
+        shipment_advices = self._plan_shipments()
+        if not shipment_advices:
+            return {}
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Shipment Advice"),
+            "view_mode": "tree,form",
+            "res_model": shipment_advices._name,
+            "domain": [("id", "in", shipment_advices.ids)],
+            "context": self.env.context,
+        }
+
+    def _get_release_channels_by_shipment_planning_method(self):
+        res = {}
+        for group in self.read_group(
+            [("id", "in", self.ids)],
+            ["shipment_planning_method"],
+            ["shipment_planning_method"],
+        ):
+            res[group.get("shipment_planning_method")] = self.search(
+                group.get("__domain")
+            )
+        return res
+
+    def _plan_shipments(self):
+        shipment_advices = self.env["shipment.advice"]
+        for (
+            shipment_planning_method,
+            channels,
+        ) in self._get_release_channels_by_shipment_planning_method().items():
+            shipment_advices |= channels._plan_shipment_for_method(
+                shipment_planning_method
+            )
+        return shipment_advices
+
+    def _get_picking_to_plan_by_warehouse(self):
+        warehouse_model = self.env["stock.warehouse"]
+        picking_model = self.env["stock.picking"]
+        res = {}
+        for group in self.picking_to_plan_ids.read_group(
+            [
+                ("release_channel_id", "in", self.ids),
+                ("planned_shipment_advice_id", "=", False),
+                ("state", "=", "assigned"),
+            ],
+            ["release_channel_id", "warehouse_id"],
+            ["release_channel_id", "warehouse_id"],
+            lazy=False,
+        ):
+            channel = self.browse(group.get("release_channel_id")[0])
+            warehouse = warehouse_model.browse(group.get("warehouse_id")[0])
+            res[channel, warehouse] = picking_model.search(group.get("__domain"))
+        return res
+
+    def _plan_shipment_for_method(self, shipment_planning_method):
```

Comment:
This method plan 0 to N shipments... `s/_plan_shipment_for_method/_plan_shipments_for_method`